core/src/main/java/org/elasticsearch/index/analysis/CustomAnalyzerProvider.java

```diff
@@ -77,13 +78,22 @@ public void build(AnalysisService analysisService) {
             tokenFilters.add(tokenFilter);
         }
 
-        int positionOffsetGap = analyzerSettings.getAsInt("position_offset_gap", 0);
-        int offsetGap = analyzerSettings.getAsInt("offset_gap", -1);
+        if (analyzerSettings.getAsMap().containsKey("position_increment_gap")
+                && analyzerSettings.getAsMap().containsKey("position_offset_gap")){
+            throw new IllegalArgumentException("Custom Analyzer [" + name() + "] has both [position_offset_gap] and [position_increment_gap]");
+        }
+
+        int positionIncrementGap = 0;
+        if (Version.indexCreated(indexSettings).before(Version.V_2_0_0) ){
```

Comment:
It gives clients an upgrade window where they can safely continue to use their old mappings for a while. Its normal not to push a new version of your application in the same release window as you push Elasticsearch. In fact I'd wag my finger accusatorially at anyone that did. So you want the old code to work with the new Elasticsearch. And the only way to rig that is to support both names for a while. Either we add support for the new name in 1.7.X and re-release or we support the old name for a couple of releases.